src/Shared/CustomPagination/CustomPagination.tsx

```diff
@@ -1,81 +1,110 @@
+              <a className="page-link">{page}</a>
+            </li>
+          </button>
+        ))}
 
-    return (
-        <nav aria-label="..." className=''>
-            <ul className="pagination justify-content-center pagination-sm">
-                {currentPage > 1 && (
-                    <li className="page-item" onClick={() => onPageChange(currentPage - 1)}>
-                        <a className="page-link">Prev</a>
-                    </li>
-                )}
+        {currentPage < totalPages && (
+            <button className="paginate btn-hover-custom bg-white p-0 m-0 border-0" onClick={() => onPageChange(currentPage + 1)}>
+                  <li
+            className="page-item"
+           
+          >
+            <a className="page-link">Next</a>
+          </li>
+            </button>
+        
+        )}
+      </ul>
 
-                {pages.map((page) => (
-                    <li key={page} className={`page-item ${page === currentPage ? 'active' : ''}`} onClick={() => onPageChange(page)}>
-                        <a className="page-link">{page}</a>
-                    </li>
-                ))}
+      <div className="input-group mb-3 w-50  d-flex justify-content-center align-items-center  m-auto">
+        <input
+          type="number"
+          className="form-control custom-input h-25 rounded"
+          placeholder="Go to page"
+          value={pageNumberInput}
+          onChange={handlePageNumberInputChange}
+          onKeyPress={handlePageNumberInputKeyPress} // Add this line
+        />
+        <div className="input-group-append m-2">
+          <button
+            className="btn btn-outline-secondary px-4"
+            type="button"
+            onClick={handleGoToPage}
+          >
+            Go
+          </button>
+        </div>
+      </div>
 
-                {currentPage < totalPages && (
-                    <li className="page-item" onClick={() => onPageChange(currentPage + 1)}>
-                        <a className="page-link">Next</a>
-                    </li>
-                )}
-            </ul>
-
-            <div className="input-group mb-3 w-50  d-flex justify-content-center align-items-center  m-auto">
-                <input
-                    type="number"
-                    className="form-control custom-input h-25 rounded"
-                    placeholder="Go to page"
-                    value={pageNumberInput}
-                    onChange={handlePageNumberInputChange}
-                    onKeyPress={handlePageNumberInputKeyPress} // Add this line
-                />
-                <div className="input-group-append m-2">
-                    <button className="btn btn-outline-secondary px-4" type="button" onClick={handleGoToPage}>Go</button>
-                </div>
-            </div>
-
-            <p className="text-center">
-                Page {currentPage} of {totalPages}
-            </p>
-        </nav>
-    );
+      <p className="text-center">
```

Comment:
When initially loaded, the pagination erroneously displays "Page 1 of 0." To enhance the user experience, consider hiding the pagination while the data is still in the process of loading